lib/galaxy/tools/__init__.py

```diff
@@ -2223,6 +2222,19 @@ def handle_input(
             collection_info=collection_info,
             completed_jobs=completed_jobs,
         )
+
+        tags = incoming.get("tags", [])
+        if not isinstance(tags, list) or not all(isinstance(tag, str) for tag in tags):
+            raise exceptions.RequestParameterInvalidException("Tags must be a list of strings.")
```

Comment:
That check should happen as early as possible, as we're validating user input I'd place that right into the `create` method. At this point we've already created the jobs, so failing here is not a good option.